File: hello-world/hello.awk.js
function main() {
    print("Hello from \x1b[30mAwk\x1b[m and \x1b[33mJavaScript\x1b[m!");
}

/*/ # code visible to Awk
# one true awk doesn't accept this as a valid regex, which is why gawk is needed

BEGIN {
    print "Hello from \x1b[30mAwk\x1b[m!"
    main()
    exit
}

# code visible to JavaScript

#*/var print;/*
#*/if (print) {/*
#*/  print("Hello from \x1b[33mqjs\x1b[m!");/*
#*/} else if (this == undefined) {/*
#*/  print = console.log;/*
#*/  print("Hello from \x1b[33mdeno\x1b[m!");/*
#*/} else if (!this.display) {/*
#*/  print = console.log;/*
#*/  print("Hello from \x1b[33mnode\x1b[m!");/*
#*/} else {/*
#*/  this.console = { log: print = function(s) { display(s); newline(); }};/*
#*/  print("Hello from \x1b[33mguile\x1b[m!");/*
#*/}/*
#*/main();

File: hello-world/hello_awk.c
#include <stdlib.h>
#include <stdio.h>
#define BEGIN void main()
#define function void
#define IS_C 1
#if 0
function puts(s) { print s }
#endif

function sayHello() {
  printf("Hello from \x1b[%s\x1b[m!\n", IS_C ? "90mC" : "30mawk");
}

BEGIN {
  sayHello();
  puts("Hello from \x1b[30mawk\x1b[m and \x1b[90mC\x1b[m!");
  exit(0);
}
